### PR TITLE
Game loop on the server

### LINKED ISSUE #13: Game loop on the server
The game loop needs to run on the server and update the server state at a given interval.
Each run of the game logic loop is referred to as a Tick.

- [x] Use an ECS to run systems on each tick
- [x] Units have a 2d position, velocity and target destination.
- [x] Units has a unit type
- [x] Entities must have a globally unique id
- [x] Units can have a target destination, that is removed once they get to it.
- [x] A system for velocity -> position
- [x] A system for target destination -> velocity
- [x] A unit that stands still (velocity 0) must have its velocity removed.
- [x] Input from users must be stored, and applied on each tick
- [x] User input must not be blocked by the run of the tick
- [x] Getting current server state must return the newest server state. 
- [x] If the tick is running, requests for game state can wait for the newest result (or return the old if possible).

### PR DESCRIPTION
**Pull request by PhilipK**
Resolves #13 

This is the start of having a game loop on the server.
The gameloop runs (known as a tick) every second.
Clients can send commands (non-blocking) to the server, and they will then be applied in the next tick.

When items are created they are immediately given an uuid, and are then later applied to the world.

### CODE REVIEW

File: server/src/game/commands/game_command.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+use crate::game::game_state::Unit;
+
+#[derive(Debug)]
+pub enum GameCommand {
+    CreateUnitCommand { unit: Unit },
+    SetUnitDestinationCommand { position: (f32, f32), uuid: String },
```
Comment (nHornDK) [inline server/src/game/commands/game_command.rs:6]: maby we should create a type for cords we could use for position,destination,rotation etc

File: server/src/game/components/velocity.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+///Stores inforamtion about an entities velocity in meters/seconds.
+pub struct Velocity{
+    ///the x-axis velocity in meters/seconds.
+    pub dx : f32,
```
Comment (MikkelHornJP) [inline server/src/game/components/velocity.rs:5]: should prob be decimal and maybe velocity should be commen for both x and y

Comment (MikkelHornJP) [reply server/src/game/components/velocity.rs:5]: also should we have an acceleration?

**Review by MikkelHornJP — APPROVED** [review]
looks good wrote a few comments but nothing that blocks

**Review by nHornDK — APPROVED** [review]

File: server/src/game/commands/game_command.rs (hunk 1 above)
Comment (PhilipK) [reply server/src/game/commands/game_command.rs:6]: Agreed, I got the vector2 library that we can use, but I wanted to make theese changes without changing the client api. A future optimization.

File: server/src/game/components/velocity.rs (hunk 2 above)
Comment (PhilipK) [reply server/src/game/components/velocity.rs:5]: Decimal? You meen f64 instead of f32?

Comment (PhilipK) [reply server/src/game/components/velocity.rs:5]: Acceleration would be fun to do, that would make it harder to "change direction" on long flights.

**Pull request merged**